src/main/java/se/sundsvall/supportcenter/service/mapper/CommonMapper.java

```diff
@@ -55,12 +57,12 @@ private static StyleEnum toNoteStyle(NoteType noteType) {
 	
 	/**
 	 * Method for extracting a note from pob payload if it exists
-	 * @param pobPayload
+	 * @param pobPayload the pob payload to extract note from
 	 * @return note in payload or null
 	 */
 	public static Note toNote(PobPayload pobPayload) {
 		final var memo = pobPayload.getMemo();
-		final var noteType = memo.keySet().isEmpty() ? null : NoteType.forValue(memo.keySet().stream().findFirst().orElse(null));
+		final var noteType = ofNullable(memo).orElse(emptyMap()).keySet().isEmpty() ? null : NoteType.forValue(ofNullable(memo).orElse(emptyMap()).keySet().stream().findFirst().orElse(null));
```

Comment:
Is this possible to simplify? (I.e. remove the ternary operator in order to reduce execution branches)

What about something like this (not tested):
```
		final var noteType = NoteType.forValue(Optional.ofNullable(memo)
			.map(Map::keySet)
			.map(Set::stream)
			.map(Stream::findFirst)
			.orElse(null)
			.orElse(null));
```

Reply:
I cannot see a better solution. It will cause an exception if null is passed to NoteType.forValue as above.

Reply:
Ahh...

This will work then?

```
		Optional.ofNullable(memo)
			.map(Map::keySet)
			.map(Set::stream)
			.map(Stream::findFirst)
			.map(optionalString -> NoteType.forValue(optionalString.get()))
			.orElse(null);
```